_posts/2018-04-09-synching-sia.md

```diff
@@ -0,0 +1,131 @@
+        Standard: 10 * time.Second,
+        Dev:      20 * time.Second,
+        Testing:  3 * time.Second,
+}).(time.Duration)
+
+// sendBlkTimeout is the timeout for the SendBlocks RPC.
+sendBlkTimeout = build.Select(build.Var{
+        Standard: 10 * time.Second,
+        Dev:      30 * time.Second,
+        Testing:  4 * time.Second,
+}).(time.Duration)
+
+// sendBlocksTimeout is the timeout for the SendBlocks RPC.
+sendBlocksTimeout = build.Select(build.Var{
+        Standard: 10 * time.Second,
+        Dev:      40 * time.Second,
+        Testing:  5 * time.Second,
+}).(time.Duration)
+
+```
+
+## Sync to a RAMDisk
+
+Validating and Applying the transactions in Sia’s blockchain is a very I/O intensive process. To reduce the amount of time spent doing this I/O I created a 11GB RAMDisk and synced the blockchain to the RAMDisk.
+
+Command to create an 11 GB RAMDisk and mount it at `Sia/consensus`.
+
+```shell
+root@v132:~# mount -t tmpfs -o size=11000m tmpfs Sia/consensus/
+root@v132:~# df -h
+Filesystem                        Size  Used Avail Use% Mounted on
+/dev/mapper/pve-vm--200--disk--1   32G  2.1G   28G   7% /
+none                              492K     0  492K   0% /dev
+tmpfs                              32G     0   32G   0% /dev/shm
+tmpfs                              32G  8.2M   32G   1% /run
+tmpfs                             5.0M     0  5.0M   0% /run/lock
+tmpfs                              32G     0   32G   0% /sys/fs/cgroup
+tmpfs                              11G     0   11G   0% /root/Sia/consensus
+```
+
+A RAMDisk has very fast I/O but is not persistent across reboots.  Most users are going to want to copy the `consensus.db` file to persistent storage after the sync is done.
+
+## Future optimizations
+
+### Increased batch size
+
+Sia downloads blocks in batches of 10. That means Sia requests blocks at least 14,800 times for a full blockchain sync. I believe Sia could request blocks in batches of 25, or even 100 and reduce some of the overhead from making so many requests. I proposed that the devs raise the batch number to 25 but my proposal was denied. You can read the discussion and compelling arguments from both sides in my pull request [#2922](https://github.com/NebulousLabs/Sia/pull/2922).
+
+### A consensus overhaul
+
+Many optimizations such as parallel block downloads, smarter timeouts, I/O buffering, and header-first block retrieval could be added with a consensus package overhaul.  I don’t believe this is the best use of time for the limited dev resources Nebulous currently has, but I do believe an overhaul will be necessary if Sia and its blockchain continue to grow.
+
+Concerns about the blockchain size and performance have been brought up several times, and the Sia devs have several ideas on how to support a large blockchain.  You can read some of the discussion on GitHub in issue [#2897](https://github.com/NebulousLabs/Sia/issues/2897).
+
+## A 73-minute sync is mostly academic
```

Comment:
I think this section kind of undermines the article. A lot of the methods you mentioned are practical if you think of it as overclocking. Can we take this section out but move the content into the relevant sections?

Like I think the stuff about RAMDisk is relevant and should just live in that section. It avoids I/O bottlenecks but you need 11 GB of RAM free and you have to awkwardly shuffle the file around.

Same thing with the timeouts. They're practical, but you'll have to adjust them based on your connection speed. The ones you used worked for a connection of 200 Mbps.